Sources/MongoSwift/MongoError.swift

```diff
@@ -173,11 +173,36 @@ public struct WriteConcernFailure: Codable {
     /// A description of the error.
     public let message: String
 
+    /// Labels that may describe the context in which this error was thrown.
+    public let errorLabels: [String]?
+
     private enum CodingKeys: String, CodingKey {
         case code
         case codeName
         case details = "errInfo"
         case message = "errmsg"
+        case errorLabels
+    }
+
+    // TODO: can remove this once SERVER-36755 is resolved
+    public init(from decoder: Decoder) throws {
+        let container = try decoder.container(keyedBy: CodingKeys.self)
+        self.code = try container.decode(ServerErrorCode.self, forKey: .code)
+        self.message = try container.decode(String.self, forKey: .message)
+        self.codeName = try container.decodeIfPresent(String.self, forKey: .codeName) ?? ""
+        self.details = try container.decodeIfPresent(Document.self, forKey: .details)
+        self.errorLabels = try container.decodeIfPresent([String].self, forKey: .errorLabels)
+    }
+
+    // TODO: can remove this once SERVER-36755 is resolved
+    internal init(
```

Comment:
Essentially, the `codeName` is not always set in `writeError`s and `writeConcernError`s so we need to manually satisfy codable conformance. This is until SERVER fixes issues in which `codeName` is missing.

Reply:
And because we add the initializer for manual `Codable` conformance, the compiler stops synthesizing the default initializer. For that reason I kind of wish there was an `@default` or something that explicitly created the initializer. It seems really easy to accidentally make a breaking change in this way.